app/controllers/admin/users_controller.rb

```diff
@@ -35,13 +35,8 @@ class Admin::UsersController < ApplicationController
 
   # GET /admin/users
   def index
-    @courses = Course.all
     @admins = User.where(is_admin: true)
-    @staff = Hash[@courses.collect { |c| [c.id, c.staff] }]
-    @teaching_assistants = Hash[@courses.collect { |c| [c.id, c.teaching_assistants] }]
-    @students = Hash[@courses.collect { |c| [c.id, c.students] }]
-    @loners = User.all - UserCourseMembership.all.collect { |m| m.user } - @admins
-    @guests = Hash[@courses.collect { |c| [c.id, c.guests] }]
+    @loners = User.all - User.joins(:memberships) - @admins
```

Comment:
Please document what this variable is for.

Reply:
I just added a brief comment in the code. Below is the use case:

CS101 instructor adds 2 TAs: TA01 and TA02 in the course CS101. For some reason, TA01 withdraws from CS101. His record will fall into the category of users with no course membership. In the next semester, CS202 instructor adds TA01 in the course CS202. SSID will retain all current information of TA01 (login name, password, ...) and add him to CS202 as a TA.